### ISSUE TITLE
Error message not output when loading a plugin

### ISSUE THREAD

**Issue by corbinu** [eslint/eslint#6976]
**What version of ESLint are you using?**
3.3.1
**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

``` js
{
  "plugins": ["@corbinu/corbinu"]
}
```

**What did you do? Please include the actual source code causing the issue.**
I tried creating a plugin with my own config. I had an error in my plugin. However this simply displayed "ESLint couldn't find the plugin" couldn't find the plugin rather than the correct error message
**What did you expect to happen?**
I expected plugin-missing.txt to output the error message thrown by node.
**What actually happened? Please include the actual, raw output from ESLint.**

```
Oops! Something went wrong! :(

ESLint couldn't find the plugin "eslint-plugin-corbinu". This can happen for a couple different reasons:

1. If ESLint is installed globally, then make sure eslint-plugin-corbinu is also installed globally. A globally-installed ESLint cannot find a locally-installed plugin.

2. If ESLint is installed locally, then it's likely that the plugin isn't installed correctly. Try reinstalling by running the following:

    npm i eslint-plugin-corbinu@latest --save-dev

If you still can't figure out the problem, please stop by https://gitter.im/eslint/eslint to chat with the team.
```

I think the template should just be modified to include the err.message from the require try incase the issue is not the plugin not being found but instead the plugin being broken. Or this should at least be output when --debug is run

**Comment by ilyavolodin** [comment]
I'm not sure about output error message to all users, we specifically added pretty messages, because people couldn't figure out what caused an issue. But I do agree that we should output error message as part of the `--debug` log. :+1:

**Comment by corbinu** [comment]
@ilyavolodin Well what if it was simply a different pretty message if the error wasn't that the plugin couldn't be found. The whole issue here is that I couldn't figure out the issue because I was told the plugin couldn't be found when it actually couldn't be loaded which are two very different things.

**Comment by ilyavolodin** [comment]
@corbinu We could probably do a different message, but I think this is meaningful only to plugin creators (very small subset of our users). I'm definitely in favor of adding it to debug log. Separate message is fine by me, as well, but I'm not 100% sure it's necessary.

**Comment by platinumazure** [comment]
:+1: I think having the original exception type and string couldn't hurt, in the message or definitely in the debug logs. The fact that the full exception info isn't in the debug logs is a travesty :cry:

**Comment by kaicataldo** [comment]
I believe this is covered by https://github.com/eslint/eslint/issues/6874

**Comment by not-an-aardvark** [comment]
This was fixed by https://github.com/eslint/eslint/pull/8323.
